test/unit/test_quantum_computer.py

```diff
@@ -470,13 +471,32 @@ def test_run_with_parameters(client_configuration: QCSClientConfiguration):
         MEASURE(0, MemoryReference("ro")),
     ).wrap_in_numshots_loop(1000)
 
-    executable.write_memory(region_name="theta", value=np.pi)
+    executable.write_memory(region_name="theta", value=param)
     bitstrings = qc.run(executable).readout_data.get('ro')
 
     assert bitstrings.shape == (1000, 1)
     assert all([bit == 1 for bit in bitstrings])
 
 
+@pytest.mark.parametrize("param", [1j, "np.pi", ["np.pi"]])
+def test_run_with_bad_parameters(client_configuration: QCSClientConfiguration, param):
+    quantum_processor = NxQuantumProcessor(nx.complete_graph(3))
+    qc = QuantumComputer(
+        name="testy!",
+        qam=QVM(client_configuration=client_configuration),
+        compiler=DummyCompiler(quantum_processor=quantum_processor, client_configuration=client_configuration),
+    )
+    executable = Program(
+        Declare(name="theta", memory_type="REAL"),
+        Declare(name="ro", memory_type="BIT"),
+        RX(MemoryReference("theta"), 0),
+        MEASURE(0, MemoryReference("ro")),
+    ).wrap_in_numshots_loop(1000)
+
+    with pytest.raises(TypeError):
```

Comment:
We should continue using `pytest.raises(..., match="some error string")` to validate the error message too